package/gluon-hoodselector/luasrc/usr/sbin/hoodselector

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/lua
+
+local nixio = require('nixio')
+local uci = require('simple-uci').cursor()
+local hoodutil = require("hoodselector.util")
+
+-- PID file to ensure the hoodselector isn't running parallel
+local lockfile = '/var/lock/hoodselector.lock'
+local lockfd = nixio.open(lockfile, 'w', 'rw-------')
+
+if not lockfd:lock('tlock') then
```

Comment:
Sure, I can change the cron file to avoid loading the lua interpreter. This should work, I took that way of lock mechanism from the lua version of the autoupdater.

Reply:
Solve in 535b0a1

Reply:
I had revoke the pre check of the lock file. nixio temporary created a file which was disappears after program terminating. This is not happen in the new solution, because the lock mechanism just use a file descriptor lock. Means the file will always present after the first execution.